crates/packaging/src/https.rs

```diff
@@ -102,6 +120,92 @@ impl<'a> PackageMetadata<'a> {
     }
 }
 
+#[test]
+#[should_panic(expected = "MissingHttps")]
+fn url_problem_missing_https() {
+    PackageMetadata::try_from("http://example.com").unwrap();
```

Comment:
Instead of `unwrap` and `should_panic`, I'd use `assert_eq!` here:

```suggestion
    assert_eq!(PackageMetadata::try_from("http://example.com"), Err(UrlProblem::MissingHttps));
```

Same with the other tests! 😃

Reply:
Oh wow 🤦‍♂️ I tried that approach first and rust started shouting at me regarding `PartialEq` or something. I guess at some point I gave up and added the `PartialEq` but then I didn't change the unit tests back. I prefer this new version indeed, thank you @rtfeldman 🙏